modules/webhooks/app/workers/work_package_webhook_job.rb

```diff
@@ -27,11 +29,37 @@
 #++
 
 class WorkPackageWebhookJob < RepresentedWebhookJob
+  attr_reader :journal
+
+  def perform(webhook_id, resource, event_name)
+    # Handle both Journal (new) and WorkPackage (backward compatibility)
+    @journal = resource.is_a?(Journal) ? resource : nil
+    @resource = @journal ? @journal.journable : resource
+
+    super(webhook_id, @resource, event_name)
+  end
+
   def payload_key
     :work_package
   end
 
   def payload_representer_class
     ::API::V3::WorkPackages::WorkPackageRepresenter
   end
+
+  def project_id
+    # If journal present, get project_id through journable
+    # Otherwise resource is already the work package
+    journal ? resource.project_id : super
+  end
+
+  def user_for_representation
+    # Use the journal's user (the updater) if available
+    # Fall back to User.system for backward compatibility or if user deleted
```

Comment:
nitpick: I think this comment can probably be a one-liner as the behaviour should be clear from the code.

```suggestion
    # Fall back to User.system for backward compatibility or if user deleted
```